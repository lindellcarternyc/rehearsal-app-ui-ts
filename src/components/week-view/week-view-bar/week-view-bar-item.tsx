import * as React from 'react'
import { MenuItem, List, Icon
} from 'semantic-ui-react'

export interface WeekViewBarItemProps {
  dayName: string,
  dayNumber: string,
  hasRehearsals: boolean
  onClick: () => void
}

const RehearsalIndicator = (props: {hasRehearsals: boolean}): JSX.Element => {
  const icon = props.hasRehearsals ? 
    (
      <Icon name='circle' size='tiny' color='blue' />
    ) : 
    (
      <Icon name='circle' size='tiny' color='grey' inverted />
    )
  return (
    <List.Description content={icon} />
  )
}

const WeekViewBarItem = (props: WeekViewBarItemProps): JSX.Element => {
  const click = props.hasRehearsals ? props.onClick : undefined
  return (
    <MenuItem onClick={click}>
      <List style={{textAlign: 'center'}}>
        <List.Item>
          <List.Description content={props.dayName} />
          <List.Header as='h3' content={props.dayNumber} />
          <RehearsalIndicator hasRehearsals={props.hasRehearsals} />
        </List.Item>
      </List>
    </MenuItem>
  )
}

export default WeekViewBarItem