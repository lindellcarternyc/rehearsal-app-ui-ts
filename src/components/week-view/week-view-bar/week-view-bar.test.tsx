import * as React from 'react'
import * as renderer from 'react-test-renderer'

import WeekViewBar, { WeekViewBarProps } from './week-view-bar'
import { WeekViewBarItemProps } from './week-view-bar-item'

describe('WeekViewBar', () => {
  it('renders correctly with no rehearsals', () => {
    const days: WeekViewBarItemProps[] = [
      {
        dayName: 'S',
        dayNumber: '14',
        hasRehearsals: false
      },
      {
        dayName: 'M',
        dayNumber: '15',
        hasRehearsals: false
      },
      {
        dayName: 'T',
        dayNumber: '16',
        hasRehearsals: false
      },
      {
        dayName: 'W',
        dayNumber: '17',
        hasRehearsals: false
      },
      {
        dayName: 'T',
        dayNumber: '18',
        hasRehearsals: false
      },
      {
        dayName: 'F',
        dayNumber: '20',
        hasRehearsals: false
      },
      {
        dayName: 'S',
        dayNumber: '20',
        hasRehearsals: false
      }
    ]
    const props: WeekViewBarProps = { days }
    const tree = renderer.create(
      <WeekViewBar {...props} />
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with rehearsals on one day', () => {
    const days: WeekViewBarItemProps[] = [
      {
        dayName: 'S',
        dayNumber: '14',
        hasRehearsals: false
      },
      {
        dayName: 'M',
        dayNumber: '15',
        hasRehearsals: false
      },
      {
        dayName: 'T',
        dayNumber: '16',
        hasRehearsals: false
      },
      {
        dayName: 'W',
        dayNumber: '17',
        hasRehearsals: true
      },
      {
        dayName: 'T',
        dayNumber: '18',
        hasRehearsals: false
      },
      {
        dayName: 'F',
        dayNumber: '20',
        hasRehearsals: false
      },
      {
        dayName: 'S',
        dayNumber: '20',
        hasRehearsals: false
      }
    ]
    const props: WeekViewBarProps = { days }
    const tree = renderer.create(
      <WeekViewBar {...props} />
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with rehearsals on multiple days', () => {
    const days: WeekViewBarItemProps[] = [
      {
        dayName: 'S',
        dayNumber: '14',
        hasRehearsals: true
      },
      {
        dayName: 'M',
        dayNumber: '15',
        hasRehearsals: true
      },
      {
        dayName: 'T',
        dayNumber: '16',
        hasRehearsals: true
      },
      {
        dayName: 'W',
        dayNumber: '17',
        hasRehearsals: true
      },
      {
        dayName: 'T',
        dayNumber: '18',
        hasRehearsals: true
      },
      {
        dayName: 'F',
        dayNumber: '20',
        hasRehearsals: true
      },
      {
        dayName: 'S',
        dayNumber: '20',
        hasRehearsals: true
      }
    ]
    const props: WeekViewBarProps = { days }
    const tree = renderer.create(
      <WeekViewBar {...props} />
    )
    expect(tree).toMatchSnapshot()
  })
})