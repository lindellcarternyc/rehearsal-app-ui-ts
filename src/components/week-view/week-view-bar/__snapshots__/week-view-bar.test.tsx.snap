// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekViewBar renders correctly with no rehearsals 1`] = `
<div
  className="ui fluid top fixed nine item menu"
>
  <div
    className="icon item"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="chevron left icon"
    />
    
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          S
        </div>
        <h3
          className="header"
        >
          14
        </h3>
        <div
          className="description"
        >
          <i
            aria-hidden="true"
            className="grey circle tiny inverted icon"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          M
        </div>
        <h3
          className="header"
        >
          15
        </h3>
        <div
          className="description"
        >
          <i
            aria-hidden="true"
            className="grey circle tiny inverted icon"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          T
        </div>
        <h3
          className="header"
        >
          16
        </h3>
        <div
          className="description"
        >
          <i
            aria-hidden="true"
            className="grey circle tiny inverted icon"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          W
        </div>
        <h3
          className="header"
        >
          17
        </h3>
        <div
          className="description"
        >
          <i
            aria-hidden="true"
            className="grey circle tiny inverted icon"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          T
        </div>
        <h3
          className="header"
        >
          18
        </h3>
        <div
          className="description"
        >
          <i
            aria-hidden="true"
            className="grey circle tiny inverted icon"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          F
        </div>
        <h3
          className="header"
        >
          20
        </h3>
        <div
          className="description"
        >
          <i
            aria-hidden="true"
            className="grey circle tiny inverted icon"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          S
        </div>
        <h3
          className="header"
        >
          20
        </h3>
        <div
          className="description"
        >
          <i
            aria-hidden="true"
            className="grey circle tiny inverted icon"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="icon item"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="chevron right icon"
    />
    
  </div>
</div>
`;

exports[`WeekViewBar renders correctly with rehearsals on multiple days 1`] = `
<div
  className="ui fluid top fixed nine item menu"
>
  <div
    className="icon item"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="chevron left icon"
    />
    
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          S
        </div>
        <h3
          className="header"
        >
          14
        </h3>
        <div
          className="description"
        >
          <i
            aria-hidden="true"
            className="blue circle tiny icon"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          M
        </div>
        <h3
          className="header"
        >
          15
        </h3>
        <div
          className="description"
        >
          <i
            aria-hidden="true"
            className="blue circle tiny icon"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          T
        </div>
        <h3
          className="header"
        >
          16
        </h3>
        <div
          className="description"
        >
          <i
            aria-hidden="true"
            className="blue circle tiny icon"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          W
        </div>
        <h3
          className="header"
        >
          17
        </h3>
        <div
          className="description"
        >
          <i
            aria-hidden="true"
            className="blue circle tiny icon"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          T
        </div>
        <h3
          className="header"
        >
          18
        </h3>
        <div
          className="description"
        >
          <i
            aria-hidden="true"
            className="blue circle tiny icon"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          F
        </div>
        <h3
          className="header"
        >
          20
        </h3>
        <div
          className="description"
        >
          <i
            aria-hidden="true"
            className="blue circle tiny icon"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          S
        </div>
        <h3
          className="header"
        >
          20
        </h3>
        <div
          className="description"
        >
          <i
            aria-hidden="true"
            className="blue circle tiny icon"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="icon item"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="chevron right icon"
    />
    
  </div>
</div>
`;

exports[`WeekViewBar renders correctly with rehearsals on one day 1`] = `
<div
  className="ui fluid top fixed nine item menu"
>
  <div
    className="icon item"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="chevron left icon"
    />
    
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          S
        </div>
        <h3
          className="header"
        >
          14
        </h3>
        <div
          className="description"
        >
          <i
            aria-hidden="true"
            className="grey circle tiny inverted icon"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          M
        </div>
        <h3
          className="header"
        >
          15
        </h3>
        <div
          className="description"
        >
          <i
            aria-hidden="true"
            className="grey circle tiny inverted icon"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          T
        </div>
        <h3
          className="header"
        >
          16
        </h3>
        <div
          className="description"
        >
          <i
            aria-hidden="true"
            className="grey circle tiny inverted icon"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          W
        </div>
        <h3
          className="header"
        >
          17
        </h3>
        <div
          className="description"
        >
          <i
            aria-hidden="true"
            className="blue circle tiny icon"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          T
        </div>
        <h3
          className="header"
        >
          18
        </h3>
        <div
          className="description"
        >
          <i
            aria-hidden="true"
            className="grey circle tiny inverted icon"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          F
        </div>
        <h3
          className="header"
        >
          20
        </h3>
        <div
          className="description"
        >
          <i
            aria-hidden="true"
            className="grey circle tiny inverted icon"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          S
        </div>
        <h3
          className="header"
        >
          20
        </h3>
        <div
          className="description"
        >
          <i
            aria-hidden="true"
            className="grey circle tiny inverted icon"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="icon item"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="chevron right icon"
    />
    
  </div>
</div>
`;
